app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RoomsController < ApplicationController
+
+  def index
+    @user = current_user
+    @currentEntries = current_user.entries
+    myRoomIds = []
+    @currentEntries.each do |entry|
+      myRoomIds << entry.room.id
+    end
+    @anotherEntries = Entry.where(room_id: myRoomIds).where.not(user_id: @user.id).order(created_at: :desc)
+  end
+
+  def show
+    @room = Room.find(params[:id])
+    if Entry.where(user_id: current_user.id, room_id: @room.id).present?
+      @direct_messages = @room.direct_messages
+      @entries = @room.entries
+    else
+      redirect_back(fallback_location: root_path)
+    end
+  end
+
+  def create
+    @room = Room.create(name: "DM")
+    @entry1 = Entry.create(room_id: @room.id, user_id: current_user.id)
+    @entry2 = Entry.create(params.require(:entry).permit(:user_id, :room_id).merge(room_id: @room.id))
```

Comment:
`params.require(:entry).permit(:user_id, :room_id)` の箇所はストロングパラメータですので， `private` メソッドで作成した方が良いでしょう。

```suggestion
    @entry2 = @roome.entries.create(entry_params)
    
    private
    
    def entry_params
      params.require(:entry).permit(:user_id, :room_id)
    end
```